### PR TITLE
Convert right all docs header to react

### PR DESCRIPTION
**Pull request by garrensmith**
Convert the header to react and clean up the code to make it one react entry point.

### CODE REVIEW

File: app/addons/documents/rightalldocsheader.react.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  './jumptodoc.react',
+  './queryoptions/stores',
+
+  //plugins
+  "../../../assets/js/plugins/prettify"
+], function (app, FauxtonAPI, Components, Documents, Databases, React, {QueryOptionsController}, QueryActions, {JumpToDoc},
+  { queryOptionsStore }) {
+
+
+  const RightAllDocsHeader = React.createClass({
```
Comment (robertkowalski) [inline app/addons/documents/rightalldocsheader.react.jsx]: general question, just curious: any reason you don't use ES6 classes (`class RightAllDocsHeader extends React.Component {};`?

File: app/addons/documents/routes-mango.js [hunk 2]
```diff
@@ -55,6 +55,10 @@ var MangoIndexEditorAndQueryEditor = BaseRoute.extend({
     MangoActions.setDatabase({
       database: this.database
     });
+
+    /*this.setComponent("#right-header", RightAllDocsHeader, {
+      database: this.database
+    });*/
```
Comment (robertkowalski) [inline app/addons/documents/routes-mango.js]: ?

File: app/addons/documents/header/header.react.jsx [hunk 3]
```diff
@@ -10,24 +10,24 @@
...
+import IndexResultStores from '../index-results/stores';
+import IndexResultsActions from '../index-results/actions';
+import QueryOptionsStore from '../queryoptions/stores';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+import { Button, ButtonGroup } from 'react-bootstrap';
 
-var ToggleHeaderButton = ReactComponents.ToggleHeaderButton;
+const {indexResultsStore} = IndexResultStores;
+const {queryOptionsStore} = QueryOptionsStore;
+const {ToggleHeaderButton} = Components;
```
Comment (benkeen) [inline app/addons/documents/header/header.react.jsx]: spaces around `{`, `}`

File: app/addons/documents/routes-index-editor.js [hunk 4]
```diff
@@ -57,6 +58,10 @@ var IndexEditorAndResults = BaseRoute.extend({
     this.createDesignDocsCollection();
     this.addLeftHeader();
     this.addSidebar();
+
+    this.setComponent("#right-header", RightAllDocsHeader, {
+      database: this.database
+    });
```
Comment (robertkowalski) [inline app/addons/documents/routes-index-editor.js]: just one page gets the alldocsheader as far as i know, now all three routes get it?

File: app/addons/documents/routes-mango.js (hunk 2 above)
Comment (benkeen) [inline app/addons/documents/routes-mango.js]: can be dropped

File: app/addons/documents/views.js [hunk 5]
```diff
@@ -1,95 +0,0 @@
...
-  serialize: function () {
-    return {
-      database: this.database.get('id')
-    };
-  }
-});
-
-Documents.Views = Views;
-
-export default Documents;
```
Comment (robertkowalski) [inline app/addons/documents/views.js]: finally <3

File: app/addons/documents/routes-index-editor.js (hunk 4 above)
Comment (garrensmith) [inline app/addons/documents/routes-index-editor.js]: The all docs header includes the query options and the doc id search ahead. Its on the view pages and all docs page.

**Comment by benkeen** [comment]
Code looks great. Just one oddity which may be unrelated: 
1. In a database all docs page, click on a metadata page for an index
2. in the Document ID typeahead thing in the header, select a document. 
Result: you get redirected to the doc, but there's a `setState()` error. This error doesn't currently occur on master.

Also - will need a rebase with master to fix the left header ellipsis style too. 

Besides the above, +1.

**Pull request merged**